target_snowflake/__init__.py

```diff
@@ -426,11 +427,16 @@ def flush_records(stream, records_to_load, row_count, db_sync, temp_dir=None, no
                 write_record_to_file(gzipfile, records_to_load, record_to_csv_line_transformer)
 
     size_bytes = os.path.getsize(csv_file)
-    s3_key = db_sync.put_to_stage(csv_file, stream, row_count, temp_dir=temp_dir)
-    db_sync.load_csv(s3_key, row_count, size_bytes)
 
-    os.remove(csv_file)
-    db_sync.delete_from_stage(stream, s3_key)
+    if load_via_snowpipe:
+        s3_key, s3_folder_name = db_sync.upload_to_stage(csv_file, stream, row_count, temp_dir=temp_dir)
```

Comment:
whats the difference between the existing `put_to_stage` and the new `upload_to_stage` functions?
in both cases the stream needs to be uploaded to an S3 bucket. I think the snowpipe load should be refactored to accordingly.